Use gradle for java import tool

lukedirtwalker opened the pull request:
-Change structure of project to match gradle structure and add a package
-Remove .classpath and .project -> they can be generated with gradle eclipse
-Add build.gradle script

With this we no longer have to manually download the required libs and run the adapt_classpath script it is all done when calling gradle build
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23discussion_r63029810%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23discussion_r63030127%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-218775653%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-218775767%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-219110090%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-219115333%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-221187569%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-221188918%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-221196853%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23issuecomment-218775653%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22So%20what%20would%20be%20the%20way%20to%20start%20using%20project%20now%3F%20What%20additional%20packages%20do%20I%20need%20to%20install%3F%20It%20would%20be%20good%20if%20you%20could%20also%20update%20the%20doxygen%20on%20compilation%20to%20include%20this.%20These%20two%20docs%3A%5Cr%5Cn%20-%20https%3A//github.com/dimitar-asenov/Envision/blob/development/doc/pages/compilation.dox%5Cr%5Cn%20-%20https%3A//github.com/dimitar-asenov/Envision/blob/development/doc/pages/try_envision.dox%22%2C%20%22created_at%22%3A%20%222016-05-12T14%3A31%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks%20for%20taking%20the%20time%20to%20improve%20our%20build%20process%2C%20I%20appreciate%20it%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-05-12T14%3A32%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hm%20I%20don%27t%20see%20too%20many%20specific%20things%20for%20the%20java%20import%20tool%20in%20the%20compilation.dox%20file%20the%20only%20additional%20dependency%20would%20be%20gradle%2C%20but%20this%20is%20only%20required%20if%20someone%20intends%20to%20use%20the%20import%20tool%2C%20should%20I%20add%20it%20anyways%3F%22%2C%20%22created_at%22%3A%20%222016-05-13T17%3A37%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yeah%2C%20you%20should%20%28just%20mention%20that%20it%27s%20for%20the%20java%20import%20tool%29%22%2C%20%22created_at%22%3A%20%222016-05-13T17%3A58%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Cool%20thanks.%20I%27ll%20take%20this%20for%20a%20spin%20right%20now%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-05-24T07%3A26%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22installing%20%60gradle%60%20pulls%20in%20128MBs%20worth%20of%20packages%20%3AD%22%2C%20%22created_at%22%3A%20%222016-05-24T07%3A33%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Cool%2C%20it%20works%20like%20a%20charm.%20Thanks%20a%20lot.%22%2C%20%22created_at%22%3A%20%222016-05-24T08%3A11%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2034a1e444e427c16cca60a0e8feb8b015c90e4dfd%20JavaImportTool/src/main/java/org/envision/java/importtool/ASTConverter.java%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/379%23discussion_r63029810%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hmm%2C%20we%20definitely%20should%20not%20use%20%60org.envision%60%20since%20that%20domain%20exists%20and%20is%20not%20ours.%20How%20important%20is%20it%20to%20have%20a%20proper%20domain%3F%20Can%20we%20skip%20the%20%60org%60%20part%20and%20just%20start%20with%20%60envision%60%3F%22%2C%20%22created_at%22%3A%20%222016-05-12T14%3A28%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sure%20I%20just%20picked%20something%20random%20%3AD%22%2C%20%22created_at%22%3A%20%222016-05-12T14%3A29%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20JavaImportTool/src/main/java/org/envision/java/importtool/ASTConverter.java%3AL23-30%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-Pull 34a1e444e427c16cca60a0e8feb8b015c90e4dfd JavaImportTool/src/main/java/org/envision/java/importtool/ASTConverter.java 5'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/379#discussion_r63029810'>File: JavaImportTool/src/main/java/org/envision/java/importtool/ASTConverter.java:L23-30</a></b>
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Hmm, we definitely should not use `org.envision` since that domain exists and is not ours. How important is it to have a proper domain? Can we skip the `org` part and just start with `envision`?
- <a href='https://github.com/lukedirtwalker'><img border=0 src='https://avatars.githubusercontent.com/u/3530769?v=3' height=16 width=16'></a> Sure I just picked something random :D
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/379#issuecomment-218775653'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> So what would be the way to start using project now? What additional packages do I need to install? It would be good if you could also update the doxygen on compilation to include this. These two docs:
- https://github.com/dimitar-asenov/Envision/blob/development/doc/pages/compilation.dox
- https://github.com/dimitar-asenov/Envision/blob/development/doc/pages/try_envision.dox
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Thanks for taking the time to improve our build process, I appreciate it :)
- <a href='https://github.com/lukedirtwalker'><img border=0 src='https://avatars.githubusercontent.com/u/3530769?v=3' height=16 width=16'></a> Hm I don't see too many specific things for the java import tool in the compilation.dox file the only additional dependency would be gradle, but this is only required if someone intends to use the import tool, should I add it anyways?
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Yeah, you should (just mention that it's for the java import tool)
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Cool thanks. I'll take this for a spin right now :)
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> installing `gradle` pulls in 128MBs worth of packages :D
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Cool, it works like a charm. Thanks a lot.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/379?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/379?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/379'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

dimitar-asenov commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/379'>Refresh</a>.<a href='#crh-update-none'></a>

lukedirtwalker committed:
Use gradle for java import tool

-Change structure of project to match gradle structure and add a package
-Remove .classpath and .project -> they can be generated with gradle
eclipse
-Add build.gradle script

With this we no longer have to manually download the required libs and
run the adapt_classpath script it is all done when calling gradle build

dimitar-asenov commented on JavaImportTool/src/main/java/org/envision/java/importtool/ASTConverter.java:
```diff
@@ -23,7 +23,7 @@
  **
  **********************************************************************************************************************/
-package javaImportTool;
+package org.envision.java.importtool;
```
Hmm, we definitely should not use `org.envision` since that domain exists and is not ours. How important is it to have a proper domain? Can we skip the `org` part and just start with `envision`?

lukedirtwalker replied:
Sure I just picked something random :D

dimitar-asenov commented:
So what would be the way to start using project now? What additional packages do I need to install? It would be good if you could also update the doxygen on compilation to include this. These two docs:
- https://github.com/dimitar-asenov/Envision/blob/development/doc/pages/compilation.dox
- https://github.com/dimitar-asenov/Envision/blob/development/doc/pages/try_envision.dox

dimitar-asenov commented:
Thanks for taking the time to improve our build process, I appreciate it :)

lukedirtwalker commented:
Hm I don't see too many specific things for the java import tool in the compilation.dox file the only additional dependency would be gradle, but this is only required if someone intends to use the import tool, should I add it anyways?

dimitar-asenov commented:
Yeah, you should (just mention that it's for the java import tool)

lukedirtwalker committed:
Describe the gradle build in the docs

dimitar-asenov commented:
Cool thanks. I'll take this for a spin right now :)

dimitar-asenov commented:
installing `gradle` pulls in 128MBs worth of packages :D

dimitar-asenov commented:
Cool, it works like a charm. Thanks a lot.
